lightning/src/ln/channelmanager.rs

```diff
@@ -4865,6 +4874,53 @@ mod tests {
 			}
 		}
 	}
+
+	#[test]
+	fn test_notify_limits() {
+		// Check that a few cases which don't require the persistence of a new ChannelManager,
+		// indeed, do not cause the persistence of a new ChannelManager.
+		let chanmon_cfgs = create_chanmon_cfgs(3);
+		let node_cfgs = create_node_cfgs(3, &chanmon_cfgs);
+		let node_chanmgrs = create_node_chanmgrs(3, &node_cfgs, &[None, None, None]);
+		let nodes = create_network(3, &node_cfgs, &node_chanmgrs);
+
+		let mut chan = create_announced_chan_between_nodes(&nodes, 0, 1, InitFeatures::known(), InitFeatures::known());
+
+		// We check that the channel info nodes have doesn't change too early, even though we try
+		// to connect messages with new values
+		chan.0.contents.fee_base_msat *= 2;
+		chan.1.contents.fee_base_msat *= 2;
+		let node_a_chan_info = nodes[0].node.list_channels()[0].clone();
+		let node_b_chan_info = nodes[1].node.list_channels()[0].clone();
+
+		// The first two nodes (which opened a channel) should now require fresh persistence
+		assert!(nodes[0].node.await_persistable_update_timeout(Duration::from_millis(1)));
```

Comment:
Not sure if by design, but don't think this test covers the `DoPersist` case: https://i.imgur.com/7fAhVOh.png

Reply:
I'm not sure what I'm looking at there, but if not that's a bug - `await_persistable_update_timeout` is supposed to return true only if we have a reason to persist.

Reply:
Just meant that the test doesn't cover the case where we *do* want to persist after handling a channel update

Reply:
Right, yes, sorry. As I was writing the test I found another bug, so the test is finished in the last commit at #949.